lumicks/pylake/fitting/model.py

```diff
@@ -161,8 +161,24 @@ def __call__(self, independent, params):
             Values for the independent parameter
         params : Params | Dict[float | Parameter]
             Model parameters
+
+        Raises
+        ------
+        ValueError
+            if values for the independent variable are provided that are not a scalar or one
+            dimensional array or list.
+        KeyError
+            if some model parameters required for simulation are missing from the supplied
+            params dictionary.
         """
         independent = np.asarray(independent, dtype=np.float64)
+
+        if independent.ndim > 1:
+            raise TypeError(
+                "Only 0- or 1-dimensional arrays are supported for the independent variable, "
+                f"got {independent.ndim}"
+            )
+
         missing_parameters = set(self.parameter_names) - set(params.keys())
         if missing_parameters:
```

Comment:
Nitpick: I saw this and couldn't resist the suggestion. 😆 
```suggestion
        if missing_parameters := set(self.parameter_names) - set(params.keys()):
```

Reply:
Nice one! Fixing... :D